Model/Indexer/AsyncEventScope.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Aligent Consulting
+ * Copyright (c) Aligent Consulting (https://www.aligent.com.au)
+ */
+
+declare(strict_types=1);
+
+namespace Aligent\AsyncEvents\Model\Indexer;
+
+use Magento\Framework\App\ScopeInterface;
+use Magento\Framework\DataObject;
+
+/**
+ * This data object class only exists for the sake of a ScopeInterface which delegates getId()
+ * to a getName();
+ *
+ * The reason for this is that almost everything upstream (at least in magento/module-elasticsearch-7) assumes
```

Comment:
Would it be worth having the store id as part of the scope, along with the event name.

I can imagine clients may wish to see events per website/store

Reply:
Agree. But thinking about this, it does not need to be at the dimension level. For example all website / stores of the same event will have the same document structure therefore it might not need it's own index.

We could simply index a `store_id` or `website_id` along with the document which means we simply filter out in Elasticsearch just like a search query.

```
PUT /index_sales.order.created
{
  "data": {
    "id": "001"
  },
  "store_id": 1
}

PUT /index_sales.order.created
{
  "data": {
    "id": "001"
  },
  "store_id": 2
}
```

instead of having multiple indices